app.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'json'
+set :erb, escape_html: true
+
+SAVE_FILE = 'memos.json'
+
+get '/' do
+  @memos = memos
+  erb :index if @memos
+end
+
+get '/edit/:id' do
+  @memo = get_memo(:id)
+  erb :edit if @memo
+end
+
+get '/new' do
+  erb :new
+end
+
+get '/show/:id' do
+  @memo = get_memo(:id)
+  erb :show if @memo
+end
+
+def memos
+  read_json[:memos]
+end
+
+def get_memo(id)
+  memos.find { |i| i[id] == params[id].to_i }
+end
+
+def add_memo(memo)
+  memo_json = read_json
+  value = memo_json[:autoincrement].to_i + 1
+
+  memo[:id] = value
+
+  memo_json[:autoincrement] = value
+  memo_json[:memos] = memo_json[:memos] << memo
```

Comment:
```suggestion
  memo_json[:memos] << memo
```

わざわざ代入しなくても`<<`は破壊的に配列の内容を更新するはずです
（チェリー本などに載っている「参照の概念」を復習してみてください）

Reply:
指摘ありがとうございます！
修正します！

>（チェリー本などに載っている「参照の概念」を復習してみてください）

こちらも改めて読み直しておきます！

コミットID： 972e8eb04b6eac7450c3acc29735384c4c8fa75a